inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +227,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
         else:
             time.sleep(time_to_wait_before_reboot)
```

Comment:
In the `recover` method, the conditional check for `self._reboot_device` should be placed outside the `else` block to avoid unnecessary sleep when a reboot is not going to happen. [important]